mod-source-record-manager-server/src/main/java/org/folio/services/afterprocessing/AdditionalFieldsUtil.java

```diff
@@ -36,9 +40,46 @@ public final class AdditionalFieldsUtil {
 
   private static final Logger LOGGER = LogManager.getLogger();
 
+  private final static CacheLoader<Object, org.marc4j.marc.Record> parsedRecordContentCacheLoader;
+  private final static LoadingCache<Object, org.marc4j.marc.Record> parsedRecordContentCache;
+
+  static {
+    // this function is executed when creating a new item to be saved in the cache.
+    // In this case this is a MARC4J Record
+    parsedRecordContentCacheLoader =
+      new CacheLoader<>() {
+        @Override
+        public org.marc4j.marc.Record load(Object parsedRecordContent) {
+          MarcJsonReader marcJsonReader =
+            new MarcJsonReader(
+              new ByteArrayInputStream(
+                parsedRecordContent.toString().getBytes(StandardCharsets.UTF_8)));
+          if (marcJsonReader.hasNext()) {
+            return marcJsonReader.next();
+          }
+          return null;
+        }
+      };
+
+    parsedRecordContentCache = CacheBuilder.newBuilder()
+      .maximumSize(2000)
```

Comment:
2000 chosen based on average size of parsed record or some other things considered?
Here is our recommendation for memory allocation for modules:
![image](https://user-images.githubusercontent.com/25097693/167127096-637fe54c-ad16-44e5-b923-ca5748d3637b.png)
More details here: https://wiki.folio.org/pages/viewpage.action?pageId=59932680

Reply:
I had updated the JIRA with a comment on how i came to the current value. Here it is below

> After this story was implemented, some profiling was done to determine if the original cache size of 25 was adequate. While the cache was being utilized and overall the number of cache hits was greater than cache misses, increasing the cache size reduces the number of cache misses for an import of about 500 records. The cache miss count stabilized at about 1600. The new cache size to be used is 2000.

It is why i wanted to know what the maximum number of records in a chunk. Memory usage did not bump to significant level. The parsed content string will be garbage collected when the record(not MARC4J) is GC'ed. I believe when the key of the cache is gone, the value of the cache is removed eventually. So i think this value could be higher but i decided to draw a line here with my 500 import file.

Reply:
Good to know, thanks